public/app/features/dashboard/utils/publicDashboards.ts

```diff
@@ -0,0 +1,3 @@
+export const isDashboardPubliclyViewed = () => {
```

Comment:
Kinda don't like coupling public pages to the url path. But its a nice simple way to move ahead with public dashboards. Could be a small chance of issues if other teams use a `/p` route for something else? Seems pretty unlikely though

Reply:
Once we've planted the flag for the route, I think this is fine. We really have nothing to go on other than the URL the first time this function is called. Maybe @grafana/user-essentials would have some feedback for us

Reply:
@joshhunt Had some opinions about this. Since we need to hide components that are rendering in `AppWrapper`, I'm not sure what a good way would be to avoid this. Once we're within the `PublicDashboardPage` component we could determine it based on some state instead of the url. I just wanted to keep it consistent, so I made all checks use this same util function.